core/input_map.cpp

```diff
@@ -175,12 +175,92 @@ bool InputMap::event_is_action(const Ref<InputEvent> &p_event, const StringName
 		ERR_FAIL_COND_V(!E, false);
 	}
 
-	Ref<InputEventAction> iea = p_event;
-	if (iea.is_valid()) {
-		return iea->get_action() == p_action;
+	Ref<InputEventAction> input_event_action = p_event;
+	if (input_event_action.is_valid()) {
+		return input_event_action->get_action() == p_action;
 	}
 
-	return _find_event(E->get().inputs, p_event, true) != NULL;
+	return _find_event(E->get().inputs, p_event) != NULL;
+}
+
+bool InputMap::event_is_action_pressed(const Ref<InputEvent> &p_event, const StringName &p_action) const {
+
+	Map<StringName, Action>::Element *E = input_map.find(p_action);
+	if (!E) {
+		ERR_EXPLAIN("Request for nonexistent InputMap action: " + String(p_action));
+		ERR_FAIL_COND_V(!E, false);
+	}
+
+	List<Ref<InputEvent> >::Element *Eevent = _find_event(E->get().inputs, p_event);
+	if (!Eevent)
+		return false;
+	Ref<InputEvent> event = Eevent->get();
+
+	bool pressed;
+	Ref<InputEventJoypadMotion> stored_joy_motion_event = event;
+	Ref<InputEventJoypadMotion> joy_motion_event = p_event;
+	if (stored_joy_motion_event.is_valid() && joy_motion_event.is_valid()) {
+		pressed = ((joy_motion_event->get_axis_value() < 0) == (joy_motion_event->get_axis_value() < 0)) && Math::abs(joy_motion_event->get_axis_value()) >= Math::abs(stored_joy_motion_event->get_axis_value());
```

Comment:
Shouldn't this be `(joy_motion_event->get_axis_value() < 0) == (stored_joy_motion_event->get_axis_value() < 0)`? Same below

Reply:
You're right :)